src/zenml/orchestrators/utils.py

```diff
@@ -71,12 +73,22 @@ def create_tfx_pipeline(
     # We do not pass the metadata connection config here as it might not be
     # accessible. Instead it is queried from the active stack right before a
     # step is executed (see `BaseOrchestrator.run_step(...)`)
-    return tfx_pipeline.Pipeline(
-        pipeline_name=zenml_pipeline.name,
-        components=list(tfx_components.values()),
-        pipeline_root=artifact_store.path,
-        enable_cache=zenml_pipeline.enable_cache,
-    )
+    try:
+        pipeline = tfx_pipeline.Pipeline(
+            pipeline_name=zenml_pipeline.name,
+            components=list(tfx_components.values()),
+            pipeline_root=artifact_store.path,
+            enable_cache=zenml_pipeline.enable_cache,
+        )
+    except InvalidDAGError:
```

Comment:
Are we sure this InvalidDAGError is only raised in cycles?

Reply:
Yes. It gets raised in two places, one of which is impossible to occur though.